scripts/zones/West_Ronfaure/npcs/Vilatroire.lua

```diff
@@ -5,29 +5,200 @@
+    elseif questIntermediateTeamwork == QUEST_ACCEPTED then
+         -- start the event that asks them if they are ready to check
+        player:startEvent(129, 0, 2)
+    elseif questIntermediateTeamwork == QUEST_COMPLETED and questAdvancedTeamwork == QUEST_AVAILABLE and sandyFame >= 4 and player:getMainLvl() >= 10 then
+        -- they've completed the second quest, the third quest is available, and they have thee required fame,
+        -- start event that starts the next quest
+        player:startEvent(131)
+    elseif questIntermediateTeamwork == QUEST_COMPLETED and questAdvancedTeamwork == QUEST_AVAILABLE and sandyFame < 4 and player:getMainLvl() < 10 then
+        -- don't meet fame requirements
+        player:startEvent(130)
+    elseif questAdvancedTeamwork == QUEST_ACCEPTED then
+         -- start the event that asks them if they are ready to check
+        player:startEvent(129, 0, 3)
+    elseif questAdvancedTeamwork == QUEST_COMPLETED then
+        player:startEvent(130)
     else
-        player:startEvent(136) -- Default - before quests
+        player:startEvent(136)
     end
 end
 
 function onEventUpdate(player, csid, option)
+    -- csid 129 happens for both quests
+    if csid == 129 then
+        local questIntroToTeamwork = player:getQuestStatus(SANDORIA,tpz.quest.id.sandoria.INTRODUCTION_TO_TEAMWORK)
+        local questIntermediateTeamwork = player:getQuestStatus(SANDORIA,tpz.quest.id.sandoria.INTERMEDIATE_TEAMWORK)
+        local questAdvancedTeamwork = player:getQuestStatus(SANDORIA,tpz.quest.id.sandoria.ADVANCED_TEAMWORK)
+
+        -- newer versions of these quests only require a party of 2. 
+        -- older versions require all 6
+        local partySizeRequirement = 2
+
+        -- get party
+        local party = player:getParty()
+
+        -- since we loop through the party to check zone and distance, may as well check these in the same loop
+        local partySameNationCount = 0
+        local partySameRaceCount = 0
+        local partySameJobCount = 0
+
+        -- make sure the party is at least the right partySizeRequirement
+        if #party >= partySizeRequirement then
+
+            -- make sure everyone in the party is in the same zone and nearby
+            for key, member in pairs(party) do
+                if member:getZoneID() ~= player:getZoneID() or member:checkDistance(player) > 15 then
+                    -- member not in zone or range
+                    player:updateEvent(1)
+                    return
+                else
+                    -- check nation for first quest
+                    if member:getNation() == player:getNation() then
+                        partySameNationCount = partySameNationCount + 1
+                    end
+
+                    -- check race for second
+                    if     (player:getRace() == tpz.race.HUME_M or player:getRace() == tpz.race.HUME_F) and (member:getRace() == tpz.race.HUME_M or member:getRace() == tpz.race.HUME_F) then
```

Comment:
We could save a few CPU cycles (and line space) by assigning locals of the player's race (right after the party size requirement succeeds) and the member's race (inside the for loop), and then check those cached values in the if instead of calling `member:getRace()` each time:
```lua
local mRace = member:getRace()
if (pRace == tpz.race.HUME_M or pRace == tpz.race.HUME_F) and (mRace == tpz.race.HUME_M or mRace == tpz.race.HUME_F) then
```